crates/sui-core/src/authority.rs

```diff
@@ -1224,15 +1231,11 @@ impl AuthorityState {
                 error: "dry-exec is only supported on fullnodes".to_string(),
             });
         }
-        match transaction.kind() {
-            TransactionKind::ProgrammableTransaction(_) => (),
```

Comment:
previous code: if PT, pass, otherwise return error (by enumerating all cases)

new behavior: if system_tx, return error, otherwise pass (because it is implied to be a PT).

Reply:
Ok, got it